spline-shadow/persistence/src/main/scala/za/co/absa/spline/persistence/model/persistenceModel.scala

```diff
@@ -17,66 +16,116 @@
 
 package za.co.absa.spline.persistence.model
 
-case class Progress(timestamp: Long, readCount: Long, _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None) extends DocumentOption
+case class Progress(timestamp: Long, readCount: Long,
+  _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None) extends DocumentOption
 
-case class Execution(id: String, dataTypes: Seq[DataType], startTime: Option[Long], endTime: Option[Long], extra: Map[String, Any], _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None) extends DocumentOption
+case class Execution(id: String, dataTypes: Seq[DataType], startTime: Option[Long], endTime: Option[Long],
+  extra: Map[String, Any], _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None)
+  extends DocumentOption
 
-case class DataType(id: String, name: String, nullable: Boolean, childrenIds: Seq[String])
+trait DataType {
+  def id: String
+  def nullable: Boolean
+}
 
-case class Schema(attributes: Seq[Attribute])
+case class SimpleDataType(
+    id: String,
+    nullable: Boolean,
```

Comment:
Let's adopt the rule to always use `override` keyword when we implement or override something, even on attributes. Although Scala compiler doesn't require it I found it very helpful during refactoring and code reading.

Reply:
added override